src/form-elements/FormElementGoogleAddress.tsx

```diff
@@ -0,0 +1,209 @@
+import * as React from 'react'
+import { FormTypes, GoogleTypes } from '@oneblink/types'
+import { useJsApiLoader } from '@react-google-maps/api'
+
+import AutocompleteDropdown from '../components/renderer/AutocompleteDropdown'
+import FormElementLabelContainer from '../components/renderer/FormElementLabelContainer'
+import useGoogleMapsApiKeyKey from '../hooks/useGoogleMapsApiKey'
+import useIsMounted from '../hooks/useIsMounted'
+import useElementAriaDescribedby from '../hooks/useElementAriaDescribedby'
+import { FormElementValueChangeHandler, IsDirtyProps } from '../types/form'
+import { OneBlinkAppsError, Sentry } from '@oneblink/apps'
+
+type Props = {
+  formId: number
+  id: string
+  element: FormTypes.GoogleAddressElement
+  value: GoogleTypes.GoogleMapsAddress | undefined
+  displayValidationMessage: boolean
+  validationMessage: string | undefined
+  onChange: FormElementValueChangeHandler<GoogleTypes.GoogleMapsAddress>
+} & IsDirtyProps
+
+function FormElementGoogleAddress({
+  id,
+  element,
+  value,
+  displayValidationMessage,
+  validationMessage,
+  onChange,
+  isDirty,
+  setIsDirty,
+}: Props) {
+  const ariaDescribedby = useElementAriaDescribedby(id, element)
+  const isMounted = useIsMounted()
+  const [label, setLabel] = React.useState('')
+  const [error, setError] = React.useState<Error | undefined>()
+  const [isLoadingAddressDetails, setIsLoadingAddressDetails] =
+    React.useState(false)
+
+  const googleMapsApiKey = useGoogleMapsApiKeyKey()
+
+  const { isLoaded } = useJsApiLoader({
+    googleMapsApiKey: googleMapsApiKey ?? '',
+    libraries: ['maps', 'marker', 'places'],
+  })
+
+  const autocompleteService = React.useMemo(() => {
+    if (isLoaded) {
+      return new google.maps.places.AutocompleteService()
+    }
+  }, [isLoaded])
+
+  const handleSearch = React.useCallback(
+    async (input: string, abortSignal: AbortSignal) => {
+      setError(undefined)
+      if (autocompleteService) {
+        const predictions = await new Promise<
+          google.maps.places.QueryAutocompletePrediction[]
+        >((resolve, reject) => {
+          autocompleteService.getQueryPredictions(
+            {
+              input,
+            },
+            (predictions, status) => {
+              if (
+                status !== google.maps.places.PlacesServiceStatus.OK ||
+                !predictions
+              ) {
+                reject(
+                  google.maps.places.PlacesServiceStatus.OK
+                    ? 'Google Places did not return any suggestions'
```

Comment:
Shouldn't we be returning an empty array if there are not predictions?

Reply:
ah yeah, maka sense